src/Core/Managed/Net46/Extensibility/Implementation/RichPayloadEventSource.cs

```diff
@@ -0,0 +1,268 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.Tracing;
+    using System.Globalization;
+
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.External;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    internal sealed class RichPayloadEventSource : IDisposable
+    {
+        /// <summary>RichPayloadEventSource instance.</summary>
+        public static readonly RichPayloadEventSource Log = new RichPayloadEventSource();
+        
+        /// <summary>Event source.</summary>
+        internal readonly EventSource eventSource;
+
+        /// <summary>Event provider name.</summary>
+        private const string EventProviderName = "Microsoft-ApplicationInsights";
```

Comment:
Is this the name for this event source? Can we use "Microsoft-ApplicationInsights-RichPayload"?